src/libs/fsdrv/lv_fs_cbfs.c

```diff
@@ -0,0 +1,233 @@
+#include "../../../lvgl.h"
+#ifdef LV_USE_FS_CBFS
+#include <stdio.h>
+#include <stdlib.h>
+#include "lv_fs_cbfs.h"
+#if defined(ARDUINO) && !defined(ESP32)
+    #include <avr/pgmspace.h>
+#endif
+
+typedef struct cbfs_handle {
+    const void * data;
+    size_t position;
+    size_t size;
+} cbfs_handle_t;
+static char drive = '\0';
+static bool ready_cb(lv_fs_drv_t * drv)
+{
+    LV_UNUSED(drv);
+    return true;
+}
+static void * open_cb(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode)
+{
+    LV_UNUSED(drv);
+    if(mode != LV_FS_MODE_RD) {
+        return NULL;
+    }
+    const char * sz = path;
+    if(*sz != '/' && *sz != '\\') {
+        return NULL;
+    }
+    ++sz;
+    if(*sz != '0') {
+        return NULL;
+    }
+    ++sz;
+    if(*sz != 'x' && *sz != 'X') {
+        return NULL;
+    }
+    ++sz;
+    const char * sz2 = strchr(sz, '-');
+    if(sz2 == NULL || sz2 == sz + 1) {
+        return NULL;
+    }
+    void * p = (void *)strtol(sz, NULL, 16);
```

Comment:
We will need to add an `lv_strtol` or tweak this function, as not every platform has a `strtol` (e.g. MicroPython on STM32). I'm still working on understanding what this does. The address and size are stored in the file path?

Reply:
> We will need to add an lv_strtol or tweak this function

We can add it in `lv_utils.c`.


> The address and size are stored in the file path?

I believe it's based on my idea [here](https://github.com/lvgl/lvgl/issues/3664#issuecomment-1242707330).

Reply:
Thank you, I was only skimming the emails and missed it.

Reply:
Yes, it was absolutely based on kisvegabor's idea. Some of the code could be commented on reflection. I parse so much it's instinct but I forget commenting it for people who don't!